addons/html_builder/static/src/plugins/header_navbar_option.js

```diff
@@ -0,0 +1,37 @@
+import { BaseOptionComponent } from "@html_builder/core/utils";
+import { onWillStart } from "@odoo/owl";
+import { rpc } from "@web/core/network/rpc";
+
+export class HeaderNavbarOption extends BaseOptionComponent {
+    static template = "html_builder.HeaderNavbarOption";
+    setup() {
+        super.setup();
+        this.activeRecords = {};
+        this.resolves = {};
+        this.pendingViewRequests = new Set();    
+        this.keys = [
+            'website.template_header_default',
+            'website.template_header_hamburger',
+            'website.template_header_boxed',
+            'website.template_header_stretch',
+            'website.template_header_vertical',
+            'website.template_header_search',
+            'website.template_header_sales_one',
+            'website.template_header_sales_two',
+            'website.template_header_sales_three',
+            'website.template_header_sales_four',
+            'website.template_header_sidebar'
+        ];
+        this.currentKey = '';
+        onWillStart(async () => {
+            return rpc("/website/theme_customize_data_get", {
+                keys: this.keys,
+                is_view_data: true,
+            }).then((r) => {
+                if (r) {
+                    this.currentKey = r[0];
+                }
```

Comment:
You should find a way to use _loadBatchKey. To use the cache and don t trigger a duplicate RPC at each time you cilck on the navbar. We try to reduce the number of RPC. Each RPC has a cost so if we can remove it is cool :)
Maybe try it in a second commit